modules/core/src/smithy4s/schema/OperationSchema.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+package schema
+
+import smithy4s.Errorable
+
+final case class OperationSchema[I, E, O, SI, SO](
```

Comment:
Start review here. This has the responsibility of carrying schemas associated to an operation.

Reply:
it's very cool

Reply:
should it have a private constructor for bincompat?

Reply:
You mean to future-proof it ? I suppose we should.

Reply:
that's what I meant yep